docs/source/tokenizer_summary.rst

```diff
@@ -1,223 +1,243 @@
+<preprocessing>`, tokenizing a text is splitting it into words or subwords, which then are converted to ids through a
+look-up table. Converting words or subwords to ids is straightforward, so in this summary, we will focus on splitting a
+text into words or subwords (i.e. tokenizing a text). More specifically, we will look at the three main types of
+tokenizers used in 🤗 Transformers: :ref:`Byte-Pair Encoding (BPE) <byte-pair-encoding>`, :ref:`WordPiece <wordpiece>`,
+and :ref:`SentencePiece <sentencepiece>`, and show exemplary which tokenizer type is used by which model.
 
-Note that on each model page, you can look at the documentation of the associated tokenizer to know which of those
-algorithms the pretrained model used. For instance, if we look at :class:`~transformers.BertTokenizer`, we can see it's
-using :ref:`WordPiece <wordpiece>`.
+Note that on each model page, you can look at the documentation of the associated tokenizer to know which tokenizer
+type was used by the pre-trained model. For instance, if we look at :class:`~transformers.BertTokenizer`, we can see
+that the model uses :ref:`WordPiece <wordpiece>`.
 
-Introduction to tokenization
+Introduction
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-Splitting a text in smaller chunks is a task that's harder than it looks, and there are multiple ways of doing it. For
-instance, let's look at the sentence "Don't you love 🤗 Transformers? We sure do." A first simple way of tokenizing this
-text is just to split it by spaces, which would give:
+Splitting a text into smaller chunks is a task that is harder than it looks, and there are multiple ways of doing so.
+For instance, let's look at the sentence "Don't you love 🤗 Transformers? We sure do." A simple way of tokenizing this
+text is to split it by spaces, which would give:
 
 .. code-block::
 
     ["Don't", "you", "love", "🤗", "Transformers?", "We", "sure", "do."]
 
-This is a nice first step, but if we look at the tokens "Transformers?" or "do.", we can see we can do better. Those
-will be different than the tokens "Transformers" and "do" for our model, so we should probably take the punctuation
-into account. This would give:
+This is a sensible first step, but if we look at the tokens "Transformers?" and "do.", we notice that the punctuation
+is attached to the words "Transformer" and "do", which is suboptimal. We should take the punctuation into account so
+that a model does not have to learn a different representation of a word and every possible punctuation symbol that
+could follow it, which would explode the number of representations the model has to learn. Taking punctuation into
+account, tokenizing our exemplary text would give:
 
 .. code-block::
 
-    ["Don", "'", "t", "you", "love", "🤗", "Transformers", "?", "We", "sure", "do", "."]
+    ["Don", """, "t", "you", "love", "🤗", "Transformers", "?", "We", "sure", "do", "."]
```

Comment:
If I'm not mistaken you got carried with a `:%s/'/"/g` :smile:

Reply:
that's very correct :D - corrected it